### PR TITLE
Implemented cucumberOpts.compiler for spec files

### PR DESCRIPTION
**Pull request by wvankuipers**
This fix will register all compilers defined in the cucumberOpts

### CODE REVIEW

File: lib/adapter.js [hunk 1]
```diff
@@ -84,6 +86,13 @@ class CucumberAdapter {
 
         return result
     }
+    
+    registerCompilers () {
```
Comment (christian-bromann) [inline lib/adapter.js]: Shouldn't this also apply for all require files?

**Comment by christian-bromann** [comment]
Maybe something like https://github.com/webdriverio/wdio-mocha-framework/blob/master/lib/adapter.js#L198-L210

**Comment by wvankuipers** [comment]
Sure thing, I will work something out

File: test/fixtures/es6.feature [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Feature: Example feature
+  As a test script of wdio-cucumber-framework
+  I should pass
+
+  Scenario: Foo Bar Doo
+    And   I go on the website "http://webdriver.io"
```
Comment (lgandecki) [inline test/fixtures/es6.feature]: Please improve the feature, it should say what you actually do test here

File: test/fixtures/es6-definition.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// import assert from 'assert'
```
Comment (lgandecki) [inline test/fixtures/es6-definition.js]: please remove if not used

File: test/fixtures/es6-definition.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+import {defineSupportCode} from 'cucumber'
+
+global.syncAsync = {}
+
+defineSupportCode(function ({Given, When, Then}) {
```
Comment (lgandecki) [inline test/fixtures/es6-definition.js]: maybe try importing given directly from cucumber, it will have a nicer syntax

**Comment by lgandecki** [comment]
Thanks for the fix, please clean it up a bit and we can merge it :)

File: test/fixtures/es6.feature (hunk 2 above)
Comment (christian-bromann) [inline test/fixtures/es6.feature]: I think it is like all other feature file just a fixture

**Pull request closed without merging**